modules/blockbuilder/blockbuilder.go

```diff
@@ -27,6 +29,35 @@ const (
 	ConsumerGroup           = "block-builder"
 )
 
+var (
+	metricPartitionLag = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "tempo",
+		Subsystem: "block_builder",
+		Name:      "partition_lag",
+		Help:      "Lag of a partition.",
+	}, []string{"partition"})
+	metricConsumeCycleDuration = promauto.NewHistogram(prometheus.HistogramOpts{
+		Namespace:                   "tempo",
+		Subsystem:                   "block_builder",
+		Name:                        "consume_cycle_duration_seconds",
+		Help:                        "Time spent consuming a full cycle.",
+		NativeHistogramBucketFactor: 1.1,
```

Comment:
do we want to limit the max buckets by setting NativeHistogramMaxBucketNumber to something high so there is a bound.

will this work for folks who are not using Prometheus v2.40+?